### DISCUSSION TITLE
Cannot start elastalert with args `--es_debug_trace trace.log`

### DISCUSSION THREAD

**Discussion by sheng-jie** [jertel/elastalert2#1086] [Q&A] [answered]
Elastalert cannot start properly when I use helm chart to deploy elastalert2 with `args: ["--verbose", "--es_debug","--es_debug_trace trace.log"` . I got the following error.

`Reading Elastic 8 index mappings:
Reading index mapping 'es_mappings/8/silence.json'
Reading index mapping 'es_mappings/8/elastalert_status.json'
Reading index mapping 'es_mappings/8/elastalert.json'
Reading index mapping 'es_mappings/8/past_elastalert.json'
Reading index mapping 'es_mappings/8/elastalert_error.json'
Index elastalert already exists. Skipping index creation.
usage: elastalert [-h] [--config CONFIG] [--debug] [--rule RULE]
                  [--silence SILENCE] [--silence_qk_value SILENCE_QK_VALUE]
                  [--start START] [--end END] [--verbose] [--patience TIMEOUT]
                  [--pin_rules] [--es_debug] [--es_debug_trace ES_DEBUG_TRACE]
                  [--prometheus_port PROMETHEUS_PORT]
elastalert: error: unrecognized arguments: --es_debug_trace trace.log`

After research, it may be caused by the insufficient permission assignment in the Dockerfile.

---

FROM python:3-slim-buster as builder

LABEL description="ElastAlert 2 Official Image"
LABEL maintainer="Jason Ertel"

COPY . /tmp/elastalert

RUN mkdir -p /opt/elastalert && \
    cd /tmp/elastalert && \
    pip install setuptools wheel && \
    python setup.py sdist bdist_wheel

FROM python:3-slim-buster

**ARG GID=1000
ARG UID=1000
ARG USERNAME=elastalert**

COPY --from=builder /tmp/elastalert/dist/*.tar.gz /tmp/
RUN apt update && apt -y upgrade && \
    apt -y install jq curl gcc libffi-dev && \
    rm -rf /var/lib/apt/lists/* && \
    pip install /tmp/*.tar.gz && \
    rm -rf /tmp/* && \
    apt -y remove gcc libffi-dev && \
    apt -y autoremove && \
    mkdir -p /opt/elastalert && \
    echo "#!/bin/sh" >> /opt/elastalert/run.sh && \
    echo "set -e" >> /opt/elastalert/run.sh && \
    echo "elastalert-create-index --config /opt/elastalert/config.yaml" \
        >> /opt/elastalert/run.sh && \
    echo "elastalert --config /opt/elastalert/config.yaml \"\$@\"" \
        >> /opt/elastalert/run.sh && \
    **chmod +x /opt/elastalert/run.sh && \
    groupadd -g ${GID} ${USERNAME} && \
    useradd -u ${UID} -g ${GID} -M -b /opt -s /sbin/nologin \
        -c "ElastAlert 2 User" ${USERNAME}**

**USER ${USERNAME}**
ENV TZ "UTC"

WORKDIR /opt/elastalert
ENTRYPOINT ["/opt/elastalert/run.sh"]

---

When I set `USERNAME=root` it works properly.

**Comment by jertel** [comment, marked as answer]
The last arg needs to be split into two:

```
args: ["--verbose", "--es_debug","--es_debug_trace", "trace.log"]
```

**Reply by sheng-jie** [reply]
It didn't work. 
![](https://user-images.githubusercontent.com/12607748/213721254-a888d082-497c-4190-8968-08c34e7bdb28.png)

**Reply by jertel** [reply]
You will need to give it a log file path that the _elastalert_ user has sufficient permissions. Ex: `/tmp/trace.log`.

**Reply by sheng-jie** [reply]
Thanks, it works.
